src/webviews/utils/ErrorBoundary.tsx

```diff
@@ -0,0 +1,81 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Link, makeStyles } from "@fluentui/react-components";
+import React from "react";
+import { useQueryEditorDispatcher } from "../QueryEditor/state/QueryEditorContext";
+
+// Error boundary component must be a class component in order to catch errors (with componentDidCatch and
+// getDerivedStateFromError). Wrap this with a functional component to access the useQueryEditorDispatcher hook.
+
+export const ErrorBoundary: React.FC<{ style?: React.CSSProperties, children?: React.ReactNode }> = ({ style, children }) => {
+    const dispatcher = useQueryEditorDispatcher();
+    return <ErrorBoundaryComponent style={style} onError={(message, stack, componentStack) =>
+        // If rendering throws right away, dispatcher.reportError might not be initialized, yet, so check first.
+        dispatcher.reportError && void dispatcher.reportError(message, stack, componentStack)} children={children} />;
+}
+
+const useStyles = makeStyles({
+    container: {
+        textAlign: 'center',
+    },
+    details:
+    {
+        textAlign: 'left',
+    }
+});
+
+const ErrorDisplay: React.FC<{ message: string | undefined, details: string | null | undefined }> = ({ message, details }) => {
+    const styles = useStyles();
+    return <div className={styles.container}>
+        <h1>An unexpected error occurred</h1>
+        <p>Please try again. If the error persists, please <Link href="https://github.com/microsoft/vscode-cosmosdb/issues/new/choose">report the issue</Link></p>
```

Comment:
Since at this point we have probably logged the error already, we should be able to reuse the report a problem command which would create the GH issue and add the error details with other metadata to it.

Reply:
Wired the "azureDatabases.reportIssue" command.